tests/drivers/adc/adc_api/boards/stm32u5a9j_dk.overlay

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	zephyr,user {
+		/* adjust channel number according to pinmux in board.dts */
+		io-channels = <&adc1 19>;
+	};
+};
+
+&adc1 {
+	/* adc1 19th channel is connected to die temperature sensor */
+	channel@19 {
+		reg = <19>;
+		zephyr,gain = "ADC_GAIN_1";
+		zephyr,reference = "ADC_REF_INTERNAL";
+		zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
+		zephyr,resolution = <14>;
+	};
+};
```

Comment:
The ADC driver alone is not suited to read the die temperature (there is a specific bit to set that this driver don't manage, only the temp driver does).
I think you don't need to define any new channels in this overlay since there are already some available from the dts.